src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -695,5 +697,41 @@ private void PersistOnShutdown()
         }
 
         #endregion
+        
+        public IReadOnlyTrieStore AsReadOnly(IKeyValueStore readOnlyStore)
```

Comment:
this is strange -> you modify the interface from a KeyValueStore to TrieStore and call it AsReadOnly
has to be changed